code/game/objects/items/weapons/RCD.dm

```diff
@@ -392,6 +419,10 @@ GLOBAL_LIST_INIT(rcd_door_types, list(
 		var/turf/AT = A
 		AT.ChangeTurf(/turf/simulated/floor/plating) // Platings go under windows.
 		return 1
+	else

```

Comment:
All of these `else`'s you've added are pointless. Just have the to_chat and play_sound where the `return 0` was before. The code was designed to work well in a early return system already.

Reply:
Sorry, didn't realise they weren't necessary. I'll go ahead and update the code to remove all the `elses` I added in and instead have it go straight to the `to_chat` and `play_sound` section before returning `false` or `0`.

I'll test the whole thing again afterwoods before pushing the updated code here.

Reply:
👍

Reply:
Your second example is good.